### PR TITLE
Strange Reagent is now Riskier

### PR DESCRIPTION
**Pull request by Crazylemon64**
Strange reagent now has a volume floor of 1 unit per dose, does 50 cloneloss on use on a dead person, and has a 10% chance upon revival of giving the revived patient a disease.

Hopefully, this should push medical to either not let patients die, or stick to more constrained/difficult, yet less punishing means of revival, such as the defib, or botany.

:cl:Crazylemon
experiment: Strange reagent now has a volume floor requirement to activate, induces clone damage, and may cause further trouble...
/:cl:

### CODE REVIEW

**Comment by SamHPurp** [comment]
I agree that SR needs some tweaking, I'm unsure if that is the right way to go about it though.

**Comment by KasparoVy** [comment]
I think it'll be an interesting change. I'm totally down to see how it fares.
It's what the experiment tag's for

**Comment by FlattestGuitar** [comment]
I'd rather see cloning harder or more time-consuming rather than that much more punishing to the person who died.

I really dislike the RNG there, though. A 10% chance that your body gets gibbed is pretty bad.

**Comment by Necaladun** [comment]
The 10% burst chance I find really interesting. It makes it a last resort or emergency thing, rather than a default.

(I'd be fine with just removing it altogether though, tbh)

**Comment by SamHPurp** [comment]
What about a 50% chance to do 150 brute? Coin flips reviving, or requiring a load more health meds.

**Comment by uraniummeltdown** [comment]
The gibbing is dumb, just make the threshold for gibbing lower and maybe even include Tox and Oxy damage. The cloning I don't like but it's tolerable

**Comment by Dinarzad** [comment]
I think you guys -strongly- overestimate the power of cloners.
SR/Defibs are better then cloning at reviving people in every way, Cloning is literally only used when you have to brain transplant someone or the other 2 methods are unavailable.
It also takes time, meaning bodies can build up faster then you can fix them/Gives antags a chance to abort the process before it's ready/a thing to bomb.
This is sort of one of those things that should kinda be taken in steps. If cloning proves to be a serious problem even after this, then go ahead and change it.
Cloning existed before SR/Defibs and it was just fine, it's really SR/Defibs and Cryo that make Medical baby-proof.

**Comment by Necaladun** [comment]
The advantage of cloning is a new body - if someone comes in with a huge amount of broken bones, damage, internal bleeding, missing limbs, etc, then cloning is the best option still.

If people can be cloned once without any problems, that's good. Multiple deaths are much rarer and some consequence for being cloned beyond the first is good.

**Comment by Twinmold93** [comment]
Cloning is a really powerful method, and at this time removes all negatives effects (any anything else can be removed via mutadone), making someone just able to be revived, regardless of time delay between death and cloning, with little or no side effects. Strange Reagent was just a faster method to get people with only a few issues, if none, to be revived quickly: Many people use it as a very fast method of revival, though is very useful for the cases of the species that CAN'T be cloned normally. Defibs make keeping a patient out of the reaches of death much easier, though they have only three minutes between defibs to do so again or stabilize the patient. It used to be as long as they weren't a skeleton you could defib them: Defibs were nerfed to make death more problematic, and of course it saw an obvious increase in the use of the cloner.

This change will make death much more costly, and may even start getting people to consider trying to avoid situations that'll kill them if they know they'll have side effects from dying.

**Comment by Fox-McCloud** [comment]
Cloning is already super time consuming unless it's been upgraded; the real issue is being able to speed-clone with no consequences. A person who's gone through the full cloning cycle is about as much inconvenienced (if not more) than someone who's gone through defib+surgery.

The SR revival though is bad though; RNG to just fail for no reason is.....yeahhhh.

If anything just make SR respect `NO_SCAN` or something like that.

**Comment by SamHPurp** [comment]
Respecting `NO_SCAN` would mean that outside of a quick Defib, Vox cannot be revived. 👎

**Comment by davidchan13** [comment]
Really the only thing about cloning that needs a nerf is speed cloning/shutting down the cloner so it force ejects a clone. /tg/ had the idea passed around of  the cloner not implementing organs right away, so perhaps adding in bits of code so that if you're ejected from the cloner before x% you spawn without certain organs or even limbs.

**Comment by Anticept** [comment]
If you do a PR like this, then put defib timer back at 5 minutes.

**Comment by Markolie** [comment]
https://github.com/tgstation/tgstation/pull/24106 (the cloning part) may also be of interest for this PR

**Comment by Fox-McCloud** [comment]
That's a much better way of handling it, in my opinion.

Full out cloning is very punishing for how long it takes, but once you start getting into speed cloning? Yeah, things get ridiculous. That PR resolves the issue very well, IMO.

**Comment by Anticept** [comment]
I like that PR much more. HOWEVER it doesn't solve the SR issue.

**Comment by davidchan13** [comment]
SR needs an incremental chance to gib based on damage to the body. Could do it a % chance based off total damage to the body - 50 (I.E 50 or lower is 0, 150 is 100% and 75 is 25%) so that doctors would be encouraged if not required to get the corpse into acceptable shape.

Alternatively fix the SR recipe so that it produces less per reaction and/or change the reagents needed to things not as easily mass produced.

**Comment by FalseIncarnate** [comment]
While personally I find the SR change fine, clearly that is a topic of no small debate. Making it less convienent is not a bad idea, but I don't think you should punish the victim by destroying their body if luck is not on their side.

Perhaps make SR result in "side-effects" such as mutations, viral infections, or organ damages, if you want it to be more on par with the other revival methods.

The cloner change is interesting (definitely encourages science to upgrade cloners more often), but as pointed out it might be too harsh on the victim. The TG cloning PR mentioned above sounds like a fair change; quantity or quality.

**Comment by Anasari** [comment]
While I am fine with cloning change (Although I prefer the option in the TG PR), I don't really like the idea of SR being dependent on RNG.

**Comment by Dinarzad** [comment]
I do like that TG PR for the cloner.
Stabs SPEEEEEEED MOOOOOOOOODE in the ass a bit, but keeps normal, slow-mode cloning as a safe alternative.

**Comment by DesolateG** [comment]
Or you could just make it so SR only works on players/simple mobs (ian?) That don't have a base SE55.

**Comment by KasparoVy** [comment]
Just had to note this, "One of the reasons cloning is so attractive is because the ORs are always backed up, and if a guy has a few broken bones, it becomes easier to just clone them." 

AND because no-one thinks to order/salvage/steal/obtain plasteel for more optables to do more surgeries per OR (could also renovate the break room as yet another double-OR, making for 6 optables).

**Comment by Iamgoofball** [comment]
Strange reagent now has a 10% chance to burst the (dead) subject, per-dose, even if they have less than 150 burn+brute.

This is retarded lmao

**Comment by SigholtStarsong** [comment]
For once, I actually agree with Goof here.

**Comment by DesolateG** [comment]
Same

**Comment by Crazylemon64** [comment]
Hmm, varying opinions on aspects of a PR? Time to split!

**Comment by Crazylemon64** [comment]
There - this PR is now just about Strange reagent. The cloning change has been moved to #6590.

**Comment by Crazylemon64** [comment]
"mass production" is meaningless when discussing SR -  a single unit may be diluted extremely, and spread among an arbitrary number of pills, and have the same effect.

To that end, I've given SR a hefty cloneloss per application (on dead people) in lieu of the %gib, and given it a volume floor, so you need pills of at least 1u instead of homeopathic levels of SR.

**Comment by Crazylemon64** [comment]
Hmm, to make things dicier, I've decided to also toss in a %chance to have it give the patient a disease (lored as "life is going wild") - something which complicates things for the reviving party as well. A simple cloneloss is not enough of a deterrent, as cryotubes are normally at hand for revivals of NOCLONE species.

**Comment by DesolateG** [comment]
Why not make SR take more to work?

**Comment by DesolateG** [comment]
And to that end, harder to make?

**Comment by Necaladun** [comment]
I like the idea especially with the lored thing, but what kinda disease here? Infectious ones etc could be...difficult.

Cloneloss is good I think! Not enough of it in the game.

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 1]
```diff
@@ -644,6 +648,21 @@
...
+						var/virus_type = pick(list(
+							/datum/disease/advance/flu,
+							/datum/disease/advance/cold,
+							/datum/disease/brainrot,
+							/datum/disease/magnitis,
+							/datum/disease/fake_gbs, // honk
+							/datum/disease/cold9
+						))
+						var/datum/disease/D = new virus_type()
+						H.AddDisease(D)
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: The grief potential for this is rather large, as are the risks to the station, just purely for bringing a single person back from the dead.

The station and medical bay shouldn't be punished for reviving someone.

Comment (Crazylemon64) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: yes they should

There are numerous means of reviving people, and each should have their own costs, consequences, and deliberations, to make death a non-trivial obstacle to medical staff. Any situation that SR can be used, for example, a body transplant and a defib can also be used - at the cost of losing the original body's "appearance", for the unclonable species.

The disease causing can be mitigated by medical anticipating this, and placing the patient into quarantine for a short period while they scout out that the patient is either clean, or needs treatment before a pandemic spreads.

The grief potential is mitigated by both the requirement for a sentient player, and a cap of 3 max chances at spawning a disease per person (before requiring a cryotrip, which is extremely non-subtle) ((100% - 90%^3) = 27.1% chance on a given person)

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: I think you're missing the point. Individual choices, on the whole should largely have individual costs, both for the person involved and the person on the receiving end of it. To that degree, all methods of revival fit within this scope. This is very different from that; it's introducing a medical-bay wide penalty) or even station wide if the person makes a mistake for the action of bringing a singular person back into the game.

Defibs work for everyone within a 5 minute window, without any negative consequences.

SR works for an unlimited duration and deals some post revive damage (albeit not much). 

Does it need tweaking? Probably so, in comparison to a defib, but externalizing the cost of going with this over another method of revival isn't the way to go about adding a downside to player agency. Individual pre and post consequences (damage from being revived, volume requirement, difficulty in acquisition, time limits, or useage requirements) are better tuning levels than group levers.

Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: Minor pedantic note: Defib is a 3 minute window (1800 deciseconds)

Comment (Crazylemon64) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: (defib time is 3 not 5 minutes)

"Individual tuning levers are better than group levers" - 
I wanna experiment with this but eh alright I'll change it to something else

Instead, I'll change it to making it revive only some of the person's non-vital limbs/organs, leaving the other ones in a state of requiring surgical attention/amputation.

Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/medicine.dm]: >Instead, I'll change it to making it revive only some of the person's non-vital limbs/organs, leaving the other ones in a state of requiring surgical attention/amputation.

Isn't that what it already does? It doesn't mend bones or internal organ damage when you use it on someone; you still have to deal with that  after they've been brought back.

**Comment by Crazylemon64** [comment]
Non-vital organs of a patient raised by SR now have a chance to be necrotic, requiring swift amputations as appropriate for the survival of the patient. The amount of organs that are rotten scales with the time dead - with 0% for sub 1-minute, and 40% for 20 minutes and more.

**Comment by Anticept** [comment]
I like the idea of necrosis. However, mitocholide is an astoundingly powerful tool to reverse dead organs and limbs, along with large amounts of charcoal and spaceacillin. I'd rather that not change, but perhaps make them less of a magic chem, to where its a lengthy process to perform.

**Comment by DesolateG** [comment]
@anticept the whole point of this is that it isn't simply *give SR* or *throw in cloner* without needing to take extra steps. This does that, and if doctors are using other meds to fix their patients them that is the intended effect. No need to further complicate it.

**Comment by Crazylemon64** [comment]
@Anticept Mitocholide doesn't fix external body parts if they're still attached to the host body

**Comment by Anticept** [comment]
@Crazylemon64 no, but it does fix organs. I'm kind of stretching beyond the scope of the PR here, but I do suspect we will be visiting mitocholide in the future in relation to SR.

**Comment by Fox-McCloud** [comment]
Defibs aren't particularly useful in their current state; unless you have a ton of brute+burn patches on you and they died very near to medbay, they're just going to quickly fall to "clone/SR them" state veryyy quickly.

**Comment by Crazylemon64** [comment]
Yeah, defibs' primary application at this point is either reviving people who died in medbay, right in front of you, or bringing frankensteins/brain transplantees back to life. Where they are is fine.

**Comment by SamHPurp** [comment]
Defibs fill a really important role right now, they seem fine to me.

**Pull request merged**